### PR TITLE
PVP-147 Add register process with Google SSO

### PR DESCRIPTION
**Pull request by klaidoshka**
- Added register process
- Added primitive profile screen to access `Sign Out` functionality
- Separated unauthenticated and authenticated scope screens with separate layouts

- Upon successful registration (sign-up) process new user model is being created.
- In case user has no accounts, app does not crash. Just informs the user about it.
- In case user has no internet, app is completely usable until user decides to sign out.

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/common/Modifiers.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import androidx.compose.foundation.background
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.core.graphics.toColorInt
+
+
+fun Modifier.backgroundGradient(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Modifiers.kt:11]: Not sure if this should stay in `Modifiers.kt` file... Maybe? If we are going to create more personal modifiers.

File: app/src/main/java/com/pvp/app/ui/screen/authentication/Components.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        verticalArrangement = Arrangement.Center
+    ) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier
+                .fillMaxHeight(0.3f)
+                .fillMaxWidth(0.75f)
+                .clip(MaterialTheme.shapes.large)
+                .background(MaterialTheme.colorScheme.primaryContainer),
+            verticalArrangement = Arrangement.Center,
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/authentication/Components.kt:58]: Noticed a comma

File: app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt [hunk 3]
```diff
@@ -71,12 +71,14 @@ fun DrawerBodyRow(
         modifier = modifier,
         verticalAlignment = Alignment.CenterVertically
     ) {
-        Icon(
-            contentDescription = route.iconDescription,
-            imageVector = route.icon
-        )
+        route.icon?.let {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt:74]: Icons are optional now.

File: app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt [hunk 4]
```diff
@@ -126,7 +128,7 @@ fun DrawerScreen(
                 .fillMaxHeight(0.1f)
         )
 
-        Divider()
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt:129]: Deprecated.

File: app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt [hunk 5]
```diff
@@ -71,12 +71,14 @@ fun DrawerBodyRow(
         modifier = modifier,
         verticalAlignment = Alignment.CenterVertically
     ) {
-        Icon(
-            contentDescription = route.iconDescription,
-            imageVector = route.icon
-        )
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt:77]: Icons are optional now.

File: app/src/main/java/com/pvp/app/ui/screen/layout/LayoutScreenAuthenticated.kt [hunk 6]
```diff
@@ -0,0 +1,188 @@
+package com.pvp.app.ui.screen.layout
+
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/layout/LayoutScreenAuthenticated.kt:3]: Unnecessary spacing

File: app/src/main/java/com/pvp/app/ui/screen/layout/LayoutScreenAuthenticated.kt [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+                        controller.navigateWithPopUp(Route.Profile.route)
+                    }
+                ) {
+                    Icon(
+                        contentDescription = "Profile screen icon",
+                        imageVector = Icons.Outlined.Person
+                    )
+                }
+            }
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/layout/LayoutScreenAuthenticated.kt:129]: Spacing

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 8]
```diff
@@ -52,10 +52,10 @@ import com.pvp.app.model.MealTask
 import com.pvp.app.model.SportActivity
 import com.pvp.app.model.SportTask
 import com.pvp.app.model.Task
+import com.yourapp.ui.components.TextField
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:55]: Not sure why we have such import... I think someone misspelled package name or used AI.

**Review by youreondrugs — APPROVED** [review]

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt (hunk 8 above)
Comment (vytjar) [reply app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:55]: Prob ChatGPT

**Review by vytjar — APPROVED** [review]
Registration and logging in seems to be working fine. Code formatting seems to be good too.

**Pull request merged**